### PR TITLE
Add set conditional contains APIs

### PR DESCRIPTION
**Pull request by PointKernel**
This PR adds conditional `contains` APIs and unifies `contains` implementations with the `contains_if_n` kernel.

### CODE REVIEW

File: include/cuco/detail/static_set/static_set.inl [hunk 1]
```diff
@@ -191,7 +191,56 @@ template <class Key,
...
+template <typename InputIt, typename StencilIt, typename Predicate, typename OutputIt>
+void static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::contains_if(
+  InputIt first,
+  InputIt last,
+  StencilIt stencil,
+  Predicate pred,
+  OutputIt output_begin,
+  cuda_stream_ref stream) const
+{
+  contains_async(first, last, stencil, pred, output_begin, stream);
```
Comment (sleeepyjack) [inline include/cuco/detail/static_set/static_set.inl]: Shouldn't this be `contains_if_async`?

File: include/cuco/detail/static_set/static_set.inl [hunk 2]
```diff
@@ -266,7 +309,7 @@ template <class Key,
           class Storage>
 static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::size_type
 static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::size(
-  cuda_stream_ref stream) const
+  cuda_stream_ref stream) const noexcept
```
Comment (sleeepyjack) [inline include/cuco/detail/static_set/static_set.inl:350]: The counter ctor might throw, so we can't use `noexcept` here.

File: include/cuco/static_set.cuh [hunk 3]
```diff
@@ -320,7 +387,7 @@ class static_set {
    * @param stream CUDA stream used to get the number of inserted elements
    * @return The number of elements in the container
    */
-  [[nodiscard]] size_type size(cuda_stream_ref stream = {}) const;
+  [[nodiscard]] size_type size(cuda_stream_ref stream = {}) const noexcept;
```
Comment (sleeepyjack) [inline include/cuco/static_set.cuh:430]: See above comment

File: include/cuco/detail/static_set/static_set.inl (hunk 1 above)
Comment (PointKernel) [reply include/cuco/detail/static_set/static_set.inl]: dang... the merge conflict got me

**Review by sleeepyjack — APPROVED** [review]

File: include/cuco/detail/static_set/static_set.inl (hunk 1 above)
Comment (sleeepyjack) [reply include/cuco/detail/static_set/static_set.inl]: Since you're going to make that last change, I'll approve the PR. Thanks!

**Pull request merged**